File: Code-By-Language/C-Sharp/March-11-2018.cs
//***************************
//*  191. Number of 1 Bits  *
//***************************

//O(1) solution [Because n is a 32-bit integer, and that's what is looped through]
public class Solution 
{
    public int HammingWeight(uint n) 
    {
        int count = 0;
        for(n = n; n != 0; n >>= 1)
        {
            if((n & 1) == 1)
            {
                count++;
            }
        }
        
        return count;
    }
}

File: Code-By-Language/Python/March-11-2018.py
#***************************
#*  191. Number of 1 Bits  *
#***************************

#O(1) solution [Because n is a 32-bit integer, and that's what is recursed through]
class Solution(object):
    def hammingWeight(self, n):
        return recurse(n,0)
    
def recurse(n,c):
    if n == 0:
        return c
    return recurse(n>>1,c+(n & 1))